dashboard/src/actions/quisbyChartActions.js

```diff
@@ -0,0 +1,124 @@
+import * as TYPES from "./types.js";
+
+import { DANGER, ERROR_MSG } from "assets/constants/toastConstants";
+
+import API from "../utils/axiosInstance";
+import { showToast } from "./toastActions";
+import { uriTemplate } from "../utils/helper";
+
+const chartTitleMap = {
+  gb_sec: "Bandwidth",
+  trans_sec: "Transactions/second",
+  usec: "Latency",
+};
+
+export const getQuisbyData = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    dispatch({
+      type: TYPES.SET_ACTIVE_RESOURCEID,
+      payload: dataset.resource_id,
+    });
+    const response = await API.get(
+      uriTemplate(endpoints, "datasets_visualize", {
+        dataset: dataset.resource_id,
+      })
+    );
+    if (response.status === 200 && response.data.json_data) {
+      dispatch({
+        type: TYPES.GET_QUISBY_DATA,
+        payload: response.data.json_data,
+      });
+      dispatch({
+        type: TYPES.IS_UNSUPPORTED_TYPE,
+        payload: "",
+      });
+      dispatch(parseChartData());
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
+    if (error?.response) {
+      const errorMsg = error.response.data.message;
+      const isUnsupportedType = errorMsg?.toLowerCase().includes("unsupported");
```

Comment:
I don't really like trying to recognize a specific error from the text of the message: among other things, it means the server can't change that message and can't use the word "unsupported" for any other error it might return.

Using "unsupported benchmark" would be a bit better; but is it really necessary for the dashboard to understand the failure instead of just reporting what the server said?

Reply:
Updated